plugins/parsers/csv/parser.go

```diff
@@ -351,9 +351,16 @@ outer:
 		}
 	}
 
-	// add metadata fields
+	// add metadata fields and tags
+outerMetadata:
 	for k, v := range p.metadataTags {
-		tags[k] = v
+		for _, tagName := range p.TagColumns {
+			if k == tagName {
+				tags[k] = v
+				continue outerMetadata
+			}
+		}
```

Comment:
Please use the `internal/choice` package here
```suggestion
		if choice.Contains(k, p.TagColumns) {
			tags[k] = v
			continue
		}
```